flink-libraries/flink-table/src/main/java/org/apache/flink/api/java/io/RowCsvInputFormat.java

```diff
@@ -0,0 +1,152 @@
+		this(filePath, DEFAULT_LINE_DELIMITER, DEFAULT_FIELD_DELIMITER, rowTypeInfo, includedFieldsMask);
+	}
+
+	public RowCsvInputFormat(Path filePath, String lineDelimiter, String fieldDelimiter, RowTypeInfo rowTypeInfo,
+			int[] includedFieldsMask) {
+		super(filePath);
+		boolean[] mask = (includedFieldsMask == null) ? createDefaultMask(rowTypeInfo.getArity())
+				: toBooleanMask(includedFieldsMask);
+		configure(lineDelimiter, fieldDelimiter, rowTypeInfo, mask);
+	}
+
+	public RowCsvInputFormat(Path filePath, RowTypeInfo rowTypeInfo, boolean[] includedFieldsMask) {
+		this(filePath, DEFAULT_LINE_DELIMITER, DEFAULT_FIELD_DELIMITER, rowTypeInfo, includedFieldsMask);
+	}
+
+	public RowCsvInputFormat(Path filePath, String lineDelimiter, String fieldDelimiter, RowTypeInfo rowTypeInfo,
+			boolean[] includedFieldsMask) {
+		super(filePath);
+		configure(lineDelimiter, fieldDelimiter, rowTypeInfo, includedFieldsMask);
+	}
+
+	private void configure(String lineDelimiter, String fieldDelimiter,
+			RowTypeInfo rowTypeInfo, boolean[] includedFieldsMask) {
+
+		if (rowTypeInfo.getArity() == 0) {
+			throw new IllegalArgumentException("Row arity must be greater than 0.");
+		}
+
+		if (includedFieldsMask == null) {
+			includedFieldsMask = createDefaultMask(rowTypeInfo.getArity());
+		}
+
+		rowSerializer = (RowSerializer) rowTypeInfo.createSerializer(new ExecutionConfig());
+
+		setDelimiter(lineDelimiter);
+		setFieldDelimiter(fieldDelimiter);
+
+		Class<?>[] classes = new Class<?>[rowTypeInfo.getArity()];
+
+		for (int i = 0; i < rowTypeInfo.getArity(); i++) {
+			classes[i] = rowTypeInfo.getTypeAt(i).getTypeClass();
+		}
+
+		setFieldsGeneric(includedFieldsMask, classes);
+	}
+
+	@Override
+	public Row fillRecord(Row reuse, Object[] parsedValues) {
+		if (reuse == null) {
+			reuse = new Row(rowSerializer.getLength());
+		}
+		for (int i = 0; i < parsedValues.length; i++) {
+			reuse.setField(i, parsedValues[i]);
+		}
+		return reuse;
+	}
+
+	@Override
+	protected boolean parseRecord(Object[] holders, byte[] bytes, int offset, int numBytes) throws ParseException {
+		boolean[] fieldIncluded = this.fieldIncluded;
+
+		int startPos = offset;
+		final int limit = offset + numBytes;
+
+		for (int field = 0, output = 0; field < fieldIncluded.length; field++) {
+
+			// check valid start position
+			if (startPos >= limit) {
+				if (isLenient()) {
+					return false;
+				} else {
+					throw new ParseException("Row too short: " + new String(bytes, offset, numBytes));
+				}
+			}
+
+			if (fieldIncluded[field]) {
+				// parse field
+				@SuppressWarnings("unchecked")
+				FieldParser<Object> parser = (FieldParser<Object>) this.getFieldParsers()[output];
+				int latestValidPos = startPos;
+				startPos = parser.parseField(bytes, startPos, limit, this.getFieldDelimiter(), holders[output]);
+				if (!isLenient() && parser.getErrorState() != ParseErrorState.NONE) {
```

Comment:
Yes, I had a look at this class as well. The difference here is that we do not want to stop parsing a line if we encounter a `ParseState.NONE` and set a `null` value instead, no?